File: 프로그래머스/js/Level 2/이진 변환 반복하기.js
function solution(s) {
  let answer = [0, 0];
  while (s != "1") {
    let oneCnt = 0;
    let zeroCnt = 0;
    for (let i = 0; i < s.length; i++) {
      if (s[i] == "1") {
        oneCnt += 1;
      } else if (s[i] == "0") {
        zeroCnt += 1;
      }
    }
    answer[1] += zeroCnt;
    s = oneCnt.toString(2);
    answer[0] += 1;
  }
  return answer;
}

File: 프로그래머스/python/Level 2/이진 변환 반복하기.py
def solution(s):
    answer = [0, 0]
    while s != "1":
        zero_cnt, one_cnt = 0, 0
        for i in range(0, len(s)):
            if s[i] == "0":
                zero_cnt += 1
            elif s[i] == "1": 
                one_cnt += 1
                
        answer[1] += zero_cnt
        s = format(one_cnt, "b")
        answer[0] += 1
            
    return answer